### PR TITLE
Changing credential management

### PR DESCRIPTION
**Pull request by sinamna**
## Summary by CodeRabbit

- **New Features**
  - Enhanced credential validation to ensure both username and password fields are present.
  - Introduced a new server configuration template for basic authentication.

- **Improvements**
  - Updated secret management to store username and password in separate entries.
  - Implemented MD5 hashing for password storage.

- **Bug Fixes**
  - Fixed an issue where the absence of a password field could cause errors in authentication.

- **Documentation**
  - Updated documentation to reflect new authentication configuration and secret management processes.

### CODE REVIEW

File: pkg/md5/md5.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"encoding/hex"
+)
+
+func MD5Hash(input string) string {
+	hasher := md5.New()
+	hasher.Write([]byte(input))
+	hashBytes := hasher.Sum(nil)
+	hashString := hex.EncodeToString(hashBytes)
+
+	return hashString
```
[the remark this answers is not in the document]
Comment (sinamna) [reply pkg/md5/md5.go:14]: doesn't have critical usage

File: internal/controller/basic_authenticator/nginx.go [hunk 2]
```diff
@@ -16,9 +18,10 @@ import (
 )
 
 const (
-	ConfigMountPath = "/etc/nginx/conf.d"
-	SecretMountDir  = "/etc/secret"
-	SecretMountPath = "/etc/secret/.htpasswd"
+	ConfigMountPath     = "/etc/nginx/conf.d"
+	SecretMountDir      = "/etc/secret"
+	SecretMountPath     = "/etc/secret/htpasswd"
+	SecretHtpasswdField = "htpasswd"
```
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/nginx.go:24]: you're right.

[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/nginx.go]: Thanks for the commit. And what do you say about moving the `const.go` file to the pkg folder? There, you can have constants defined in `random_generator.go` and `basicauthenticator_webhook.go`.

File: internal/controller/basic_authenticator/nginx.go [hunk 3]
```diff
@@ -131,6 +114,19 @@ func createNginxConfigmap(basicAuthenticator *v1alpha1.BasicAuthenticator) *core
...
+	if !ok {
+		return defaultError.New("username not found in secret")
+	}
+	password, ok := secret.Data["password"]
+	if !ok {
+		return defaultError.New("password not found in secret")
+	}
+	htpasswdString := fmt.Sprintf("%s:%s", string(username), md5.MD5Hash(string(password)))
+	secret.Data["htpasswd"] = []byte(htpasswdString)
+	return nil
```
[the remark this answers is not in the document]
Comment (hoptical) [reply internal/controller/basic_authenticator/nginx.go:128]: It's fine.

File: internal/controller/basic_authenticator/nginx.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sinamna) [reply internal/controller/basic_authenticator/nginx.go]: Im not ok with this. I think pkg should be global and reusable (thats why we separate it from internal) but moving controller specific consts into global pkg kinda mixes things up.

**Review by hoptical — APPROVED** [review]

**Pull request merged**